src/commands/app/install.js

```diff
@@ -0,0 +1,129 @@
+const execa = require('execa')
+const { validateJsonWithSchema } = require('../../lib/install-helper')
+
+class InstallCommand extends BaseCommand {
+  async run () {
+    const { args, flags } = await this.parse(InstallCommand)
+
+    aioLogger.debug(`flags: ${JSON.stringify(flags, null, 2)}`)
+    aioLogger.debug(`args: ${JSON.stringify(args, null, 2)}`)
+
+    // resolve to absolute path before any chdir
+    args.path = path.resolve(args.path)
+    aioLogger.debug(`args.path (resolved): ${args.path}`)
+
+    let outputPath = flags.output
+    // change the cwd if necessary
+    if (outputPath !== '.') {
+      outputPath = path.resolve(flags.output)
+      // TODO: confirm if dir exists for overwrite
+      await fs.ensureDir(outputPath)
+      process.chdir(outputPath)
+      aioLogger.debug(`changed current working directory to: ${outputPath}`)
+    }
+
+    await this.validateZipDirectoryStructure(args.path)
+    await this.unzipFile(args.path, outputPath)
+    await this.validateConfig(outputPath, 'app.config.yaml')
+    await this.validateConfig(outputPath, 'deploy.yaml')
+    await this.runTests(outputPath)
+  }
+
+  diffArray (left, right) {
+    if (!left || !right) {
+      return left
+    }
+
+    return left.filter(item => !right.includes(item))
+  }
+
+  async validateZipDirectoryStructure (zipFilePath) {
+    aioLogger.debug(`validateZipDirectoryStructure: ${zipFilePath}`)
+
+    const expectedFiles = ['app.config.yaml', 'deploy.yaml', 'package.json']
+    const foundFiles = []
+
+    this.log(`Validating integrity of app package at ${zipFilePath}...`)
+
+    const zip = fs.createReadStream(zipFilePath).pipe(unzipper.Parse({ forceStream: true }))
+    for await (const entry of zip) {
+      const fileName = entry.path
+
+      if (expectedFiles.includes(fileName)) {
+        foundFiles.push(fileName)
+      }
+      entry.autodrain()
+    }
+
+    const diff = this.diffArray(expectedFiles, foundFiles)
+    if (diff.length > 0) {
+      this.error(`The app package ${zipFilePath} is missing these files: ${JSON.stringify(diff, null, 2)}`)
+    }
+  }
+
+  async unzipFile (zipFilePath, destFolderPath) {
+    aioLogger.debug(`unzipFile: ${zipFilePath} to be extracted to ${destFolderPath}`)
+
+    this.log(`Extracting app package to ${destFolderPath}...`)
+    return unzipper.Open.file(zipFilePath)
+      .then(d => d.extract({ path: destFolderPath, concurrency: 5 }))
+  }
+
+  async validateConfig (outputPath, configFileName) {
+    this.log(`Validating ${configFileName}...`)
+    const { valid, errors } = validateJsonWithSchema(path.join(outputPath, configFileName), configFileName)
+    if (!valid) {
+      const message = `Missing or invalid keys in ${configFileName}: ${JSON.stringify(errors, null, 2)}`
+      this.error(message)
+    }
+  }
+
+  async runTests (outputPath) {
+    this.log('Running tests...')
+    const { exitCode } = await execa('aio', ['app', 'test'], { cwd: outputPath, stdio: 'inherit' })
```

Comment:
should we call via oclif? or it doesn't matter

Reply:
Copy and paste -- I'll see if I can swap it with an oclif call

Reply:
fixed with https://github.com/adobe/aio-cli-plugin-app/pull/662/commits/8f247fe7dd28c532a3d38a168222ef6e01796b95